src/sharding/Shard.js

```diff
@@ -117,11 +140,25 @@ class Shard extends EventEmitter {
           execArgv: this.execArgv,
         })
         .on('message', this._handleMessage.bind(this))
-        .on('exit', this._exitListener);
+        .on('exit', this._exitListener)
+        .on('disconnect', this._handleDisconnect.bind(this, { parentProcessDisconnect: true }));
+
+      this.pid = this.process.pid;
+      this.ppid = this.process.ppid;
     } else if (this.manager.mode === 'worker') {
+      this.parentPort = require('worker_threads').parentPort;
```

Comment:
Couldn't `require('worker_threads')` be stored as a constant and then read this as `workerThreads.parentPort`? The `require()` has some overhead, and in top of that, the module is cached and can never change.

Reply:
Additionally, you could remove `this.parentPort` and use that one for all the checks, removes a state layer and relies more on the source of truth.

Reply:
> Additionally, you could remove `this.parentPort` and use that one for all the checks, removes a state layer and relies more on the source of truth.

I decided on having `this.parentPort` as a thing since ShardClientUtil also exposes such a property.